src/workerd/io/worker.c++

```diff
@@ -1839,6 +1883,42 @@ void Worker::Lock::logUncaughtException(UncaughtExceptionSource source,
   }
 }
 
+void Worker::Lock::logUncaughtException(UncaughtExceptionSource source,
+                                        kj::Exception&& exception) {
+  jsg::Lock& js = *this;
+
+  // If we have an attached serialized exception, deserialize it and log that instead, rather
+  // than try to reconstruct based on the KJ exception description.
+  //
+  // TODO(cleanup): Eventually, js.exceptionToJsValue() should do this internally, and then we
+  //   should remove the code from here.
+  KJ_IF_SOME(serializedJsError, exception.getDetail(jsg::TUNNELED_EXCEPTION_DETAIL_ID)) {
+    if (!js.v8Isolate->IsExecutionTerminating()) {
+      kj::Maybe<jsg::JsValue> deserialized;
+
+      v8::TryCatch tryCatch(js.v8Isolate);
+      try {
+        jsg::Deserializer deser(js, serializedJsError);
+        deserialized = deser.readValue(js);
+      } catch (jsg::JsExceptionThrown&) {
+        // Failed to deserialize, we'll continue with exceptionToJsValue() instead.
+        //
+        // Note that we're intentionally not checking tryCatch.CanContinue() here, because we still
+        // want to log the exception even if the isolate has been terminated.
+      }
+
+      KJ_IF_SOME(d, deserialized) {
```

Comment:
Looks like `exceptionToJsValue()` does some additional property population, filtering, and logging of internal tunneled exceptions that would be skipped when deserialization is successful.  I guess the assumption here is that if a valid serialized value is present, it didn't come from an internal tunneled exception?  And that the other populated properties are not currently consumed by `logUncaughtException()`?

Reply:
Yeah, `logUncaughtException()` doesn't inspect any of those extra properties today. Only type, message, and now stack.